HA/xcatha.py

```diff
@@ -427,84 +451,150 @@ def deactivate_management_node(self, nic, vip, dbtype):
         global setup_process_msg
         setup_process_msg="Deactivate stage"
         self.log_info(setup_process_msg)
-        self.execute_command("chkconfig --level 345 xcatd off")
-        self.execute_command("chkconfig --level 2345 conserver off")
-        self.execute_command("chkconfig --level 2345 dhcpd off")
-        self.execute_command("chkconfig postgresql off")
-        self.execute_command("service conserver stop")
-        self.execute_command("service dhcpd stop")
-        self.execute_command("service named stop")
-        self.execute_command("service xcatd stop")
-        stop_db="service "+dbtype+" stop"
-        self.execute_command(stop_db)
-        self.execute_command("service ntpd restart")
-        self.unconfigure_shared_data(shared_fs)
+        #MG How do we know which original IP was used ?
+        real_ip="1.2.3.4"
+        restore_host_name=self.get_hostname_for_ip(real_ip)
+        if restore_host_name:
+            logger.info("Restoring original hostname: " + restore_host_name)
+            self.change_hostname(restore_host_name,vip)
+        else:
+            logger.info("Error: Can not restore original hostname")
         self.unconfigure_vip(vip, nic)
+        self.unconfigure_shared_data(shared_fs)
+        self.execute_command("chkconfig --level 345 xcatd off",3)
```

Comment:
I don't like this here.

1. I think we should have a generic run_command() function that is called instead of `self.execute_command()`   
2. There's no checking of the return codes of the commands 
3. We should have a file or some other data structure that we can modify to change the commands we run.  Doing it like this is a lot of work when we have to add commands, remove commands or tweak it.